### PR TITLE
[BEAM-7203] Signal Dataflow workers to use fastavro library on Python 3.

### PR DESCRIPTION
**Pull request by tvalentyn**
https://github.com/apache/beam/pull/8130 makes fastavro a default Avro implementaion for Avro Source / Sink on Python 3. This PR instructs Dataflow workers to also use fastavro by default on Python 3.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by tvalentyn** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
R: @chamikaramj 
I believe the postcommit failures are unrelated and review can proceed in parallel with obtaining a green run.

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 1]
```diff
@@ -431,6 +432,14 @@ def run_pipeline(self, pipeline, options):
       else:
         debug_options.add_experiment('use_staged_dataflow_worker_jar')
 
+    # Make Dataflow workers use FastAvro on Python 3 unless use_avro experiment
+    # is set. Note that use_avro is only interpreted by the Dataflow Runner
+    # and not interpreted by Dataflow workers, which by default use avro
+    # library unless use_fastavro experiment is setd.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: s/setd/set

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py [hunk 2]
```diff
@@ -431,6 +432,14 @@ def run_pipeline(self, pipeline, options):
       else:
         debug_options.add_experiment('use_staged_dataflow_worker_jar')
 
+    # Make Dataflow workers use FastAvro on Python 3 unless use_avro experiment
+    # is set. Note that use_avro is only interpreted by the Dataflow Runner
```
Comment (chamikaramj) [inline sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Did you mean "only interpreted by the Dataflow SDK and not interpreted by Dataflow service/workers" ?

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Updated, thanks!

File: sdks/python/apache_beam/runners/dataflow/dataflow_runner.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/dataflow_runner.py]: Thanks, done.

**Comment by tvalentyn** [comment]
Run Python Postcommit

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request merged**